aten/src/ATen/native/sparse/cuda/SparseCUDABlas.cu

```diff
@@ -9,7 +9,17 @@
 
 #include <cusparse.h>
 
-#if defined(__CUDACC__) && CUSPARSE_VERSION >= 11000
+// LIMITATION (cusparseSpMM):
+// The generic APIs are available on all platforms on CUDA 11.0
+// For CUDA 10.1+ it is available for all platforms except Windows.
+// Using these APIs in any other systems will result in compile-time or run-time failures.
+// Their support will be extended in the next releases.
+
+#define IS_SPMM_AVAILABLE() (defined(__CUDACC__) &&                              \
```

Comment:
Hey @malfet , looks like this macro caused the Windows failure. The weird thing is that if I replace `IS_SPMM_AVAILABLE()` with the content in it, it works.

Reply:
```cpp
#include <iostream>
#include <cusparse.h>

using namespace std;

#define IS_SPMM_AVAILABLE() (defined(__CUDACC__) &&                              \
                             (CUSPARSE_VERSION >= 11100 ||                       \
                              (!defined(_MSC_VER) && CUSPARSE_VERSION >= 10301)))

int main() {
    cout << CUSPARSE_VERSION << endl;
    cout << __CUDACC__ << endl;
    cout << _MSC_VER << endl;
    cout << (__CUDACC__ > 0 && CUSPARSE_VERSION >= 11000) << endl;
    cout << (_MSC_VER > 0 && CUSPARSE_VERSION >= 10301) << endl;
#if (defined(__CUDACC__) && (CUSPARSE_VERSION >= 11100 || (!defined(_MSC_VER) && CUSPARSE_VERSION >= 10301)))
    cout << "hey" << endl;
#endif
#if IS_SPMM_AVAILABLE()
    cout << "yes" << endl;
    return 0;
#else
    cout << "no" << endl;
    return 1;
#endif
}
gives the output:
```
11100
1
1926
1
1
hey
no
```
```

Reply:
I found something similar https://github.com/cisco/cjose/issues/63. 
A fix is at https://github.com/pytorch/pytorch/pull/42643.

Reply:
```cpp
#define __CUDACC__ 1

#if defined(__CUDACC__)
#error works
#endif

#if IS_CUDACC()
#error this one doesn't work
#endif
```

Reply:
@peterjc123 yeah, I keep forgetting about this, but `defined()` is not a macro that returns a value nor is it a function.